// js/index/index4/scrollabletabs.js
import React, {Component, PropTypes} from 'react';
import {View, Text, StyleSheet, Dimensions, Animated, Easing, StatusBar} from 'react-native';
import ScrollableTabs from 'comp/modules/scrollview/ScrollableTabs';

import blog from '../business/blog'

/*  sample 1 */
let normal = ( <Text>demo</Text> )

let contents = {
    good: '1',
    ask: '2',
    all: blog,
    share: '3',
    job: '4'
}

export class simpleScrollTab extends Component {
    _renderTopicList() {
		return ['good', 'ask', 'all', 'share', 'job'].map((item, i)=> {
            if (contents[item]){
                if (typeof contents[item] === 'string'){
                    return (
                        <View key={item} >
                            <Text> {contents[item]} </Text>
                        </View>
                    )
                }
                else {
                    return (
                        <View key={item} style={{flex:1}}>
                            {contents[item]}
                        </View>
                    )
                }
            }
		});
	}

    _onPageChanged(page, isScrolling) {
        console.log(page);
	}

    render() {
        return (
            <ScrollableTabs
                ref={view=> this._scrollableTabs=view}
                tabs={['精华', '问答', '主页', '分享', '招聘']}
                onPageChangedAndAnimateEnd={this._onPageChanged.bind(this)}
            >
                { this._renderTopicList() }
            </ScrollableTabs>

        )
    }
}
