// js/index/index4/tabbar.js

import React, { Component } from 'react';
import { View, Text, Image, TouchableHighlight, Platform } from 'react-native';
import styles from 'css/index';
import tabs from 'comp/modules/tabbar/test';
import Icon from 'react-native-vector-icons/Ionicons';
import {simpleScrollTab as SimpleScrollTab} from './scrolltab';

import page from 'comm/page'   //router must use this function

let router;

let activeStyle = {fontSize:11}

// tab button
let blogIcon = [
    // normal stat
    <View>
        <Icon name="ios-home-outline" size={30} color="#4F8EF7" />
        <Text style={{fontSize:11}}>博客</Text>
    </View>,

    // active stat
    <View>
        <Icon name="ios-home" size={30} color="#4F8EF7" />
        <Text style={activeStyle}>博客</Text>
    </View>
]

let msgIcon = [
    <View>
        <Icon name="ios-chatbubbles-outline" size={30} color="#4F8EF7" />
        <Text style={{fontSize:11}}>消息</Text>
    </View>,
    <View>
        <Icon name="ios-chatbubbles" size={30} color="#4F8EF7" />
        <Text style={activeStyle}>消息</Text>
    </View>
]

let setIcon = [
    <View>
        <Icon name="ios-settings-outline" size={30} color="#4F8EF7" />
        <Text style={{fontSize:11}}>设置</Text>
    </View>,
    <View>
        <Icon name="ios-settings" size={30} color="#4F8EF7" />
        <Text style={activeStyle}>设置</Text>
    </View>
]

let bdgIcon = [
    <View>
        <Icon name="ios-settings-outline" size={30} color="#4F8EF7" />
        <Text style={{fontSize:11}}>红点</Text>
    </View>,
    <View>
        <Icon name="ios-settings" size={30} color="#4F8EF7" />
        <Text style={activeStyle}>红点</Text>
    </View>
]

let rtIcon = [
    <View>
        <Icon name="ios-settings-outline" size={30} color="#4F8EF7" />
        <Text style={{fontSize:11}}>路由</Text>
    </View>,
    <View>
        <Icon name="ios-settings" size={30} color="#4F8EF7" />
        <Text style={activeStyle}>路由</Text>
    </View>
]


// tags item config

let _blog = {
    title: blogIcon,
    content: <SimpleScrollTab />,
    method:()=>{}
}

let _msg = {
    title: msgIcon,
    content: 'content为字符串会自动在tabs里生成结构',
    method:()=>{
        console.log('消息中心')
        console.log(SAX.lister());
    }
}

let _set = {
    title: setIcon,
    content: (<View>
            <Text>content为自定义结构，会自动输出</Text>
        </View>
    ),
    method:()=>{
        console.log('设置中心')
    }
}

// 小红点数据
// 通过sax设置小红点
let saxData = {
    data: {
        badge: {
            message: 3,
            setting: 2,
            badge: 11
        }
    }
}
let _bdg = {
    title: bdgIcon,
    content: '可以有badge',
    badge: 1,
    method:()=>{
        SAX.setter('TabBlog', saxData)
    }
}

// 生成tabbar
let Home = tabs({
        blog: _blog,
        message: _msg,
        setting: _set,
        badge: _bdg
    },
    // options
    { sax: 'TabBlog' }
)

export default function blog(id){
    return page.new({
        trigger: function(self, intent){
            global._router = this.router;
            return <Home />
        }
    })
}
