// css/index.js
import { StyleSheet, Platform, Dimensions } from 'react-native';

const {_height, _width} = Dimensions.get('window');

const PT = Platform.OS === 'ios' ? 20 : 0;

export default StyleSheet.create({
    hlist: {
        flexDirection: 'row',
    },
    hheader: {
        fontSize: 16,
    },

    webView: {
        backgroundColor: 'rgba(255,255,255,0.8)',
        paddingLeft: 10,
        paddingRight: 10,
        height: 550,
    },

    navBar: {
        paddingTop: PT,
        paddingLeft: 10,
    },

    goback: {
        position: 'absolute',
        bottom: 30,
        left: 30,
    },


    scrollView: {
        paddingTop: 20,
        height: 300,
        backgroundColor: '#F5FCFF',
        overflow: 'hidden'
    },
    listView: {
        backgroundColor: '#F5FCFF',
        overflow: 'hidden'
    },
            item: {
                backgroundColor: '#e8e6e6',
                // paddingLeft: 10,
                // paddingRight: 10,
                // paddingBottom: 5,
                // paddingTop: 5,
            },
                    inner: {
                        flex: 1,
                        flexDirection: 'row',
                        justifyContent: 'center',
                        alignItems: 'center',
                        backgroundColor: '#ffffff',
                        borderRadius: 5,
                        padding: 10,
                    },
                    innerBg: {
                        borderRadius: 10,
                    },

                            rightPart: {
                                flex: 1,
                                marginLeft: 10,
                            },
                            title: {
                                fontSize: 18,
                                fontWeight: 'bold',
                                marginBottom: 8,
                                textAlign: 'left',
                            },
                            content: {
                                fontSize: 18,
                                fontWeight: 'bold',
                                marginBottom: 8,
                                textAlign: 'left',
                            },
                                    year: {
                                        color: 'rgb(135, 135, 135)',
                                        marginLeft: 4,
                                        textAlign: 'left',
                                        marginTop: -5,
                                    },
                                    subTitle: {
                                        color: 'rgb(135, 135, 135)',
                                        marginLeft: 3,
                                        fontSize: 14,
                                        marginTop: 3,
                                    },
                                    dot: {
                                        width: 40,
                                        height: 40,
                                        position: 'absolute',
                                        left: 0,
                                        top: 0,
                                    },
                                    thumbnail: {
                                        width: 80,
                                        height: 80,
                                        borderRadius: 15,
                                    },

    loading: {
        flex: 1,
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center',
        padding: 10,
        backgroundColor: '#ffffff',
        marginBottom: -10,
        borderRadius: 10,
    },
});
