// css/tabbar/index.js
import { StyleSheet, Platform, Dimensions } from 'react-native';

const {_height, _width} = Dimensions.get('window');

const PT = Platform.OS === 'ios' ? 20 : 0;

export default StyleSheet.create({
    tabBarView:{
        flex: 1,
        flexDirection: 'column',
        paddingTop: PT,
        // ...Platform.select({
        //     ios: {
        //         paddingTop: 30,
        //     },
        //     android: {
        //         paddingTop: 0,
        //     },
        // }),
    },
            contents: {
                position: 'relative',
                backgroundColor: '#fff',
                flex: 10,
            },
                    content: {
                        width: _width,
                        overflow: 'hidden'
                    },
                    hidden: {
                        overflow: 'hidden',
                        position: 'absolute',
                        left: -2000,
                    },

            bars: {
                backgroundColor: '#ffffff',
                flex: 1,
                flexDirection: 'row',
                borderTopWidth: 1,
                borderTopColor: "#ccc"
            },
                    bar: {
                        position: 'relative',
                        backgroundColor: '#ffffff',
                        flex: 1,
                        alignItems: 'center',
                        justifyContent: 'center',
                    },
                    active: {
                        position: 'relative',
                        backgroundColor: '#ffffff',
                        flex: 1,
                        alignItems: 'center',
                        justifyContent: 'center'
                    },
                            badge: {
                                position: 'absolute',
                                width: 20,
                                height: 20,
                                alignItems: 'center',
                                justifyContent: 'center',
                                right: 5,
                                top: 0
                            },
                                    badgeText:{
                                        backgroundColor: 'transparent',
                                        color: '#ffffff',
                                        fontSize: 12
                                    }

})
