// css/tabbar/index.js
import { StyleSheet } from 'react-native';
export default StyleSheet.create({
    tabBarView:{
        flex: 1,
        flexDirection: 'column'
    },
            contents: {
                backgroundColor: '#fff',
                paddingTop: 30,
                flex: 10,
            },
                    content: {
                        paddingLeft: 10,
                        paddingRight: 10,
                    },

            bars: {
                position: 'relative',
                backgroundColor: '#ffffff',
                flex: 1,
                flexDirection: 'row',
                borderTopWidth: 1,
                borderTopColor: "#ccc"
            },
                    bar: {
                        backgroundColor: '#ffffff',
                        flex: 1,
                        alignItems: 'center',
                        justifyContent: 'center',
                    },
                    active: {
                        height: 60,
                        backgroundColor: '#ffffff',
                        flex: 1,
                        alignItems: 'center',
                        justifyContent: 'center'
                    },
                            badge: {
                                position: 'absolute',
                                width: 20,
                                height: 20,
                                alignItems: 'center',
                                justifyContent: 'center',
                                right: -12,
                                top: -3
                            },
                                    badgeText:{
                                        backgroundColor: 'transparent',
                                        color: '#ffffff',
                                        fontSize: 12
                                    }

})
