// comp/modules/webrender/index.js
import React, { Component } from 'react';
import { View, Text, Image, TouchableOpacity, WebView, Platform, BackAndroid } from 'react-native';
import styles from 'css/index'
import page from 'comm/page'
import Icon from 'react-native-vector-icons/Ionicons';


let _page;

const WEBVIEW_REF = 'webview'



class Aaa extends Component {
    constructor(props){
        super(props)
        this.state = {
            url: this.props.data.url,
            status: 'No Page Loaded',
            backButtonEnabled: false,
            forwardButtonEnabled: false,
            loading: true,
            scalesPageToFit: true,
        };

        this.onShouldStartLoadWithRequest = this.onShouldStartLoadWithRequest.bind(this);
        this.onNavigationStateChange = this.onNavigationStateChange.bind(this);
        this.webGoBack = this.webGoBack.bind(this)

    }

    onShouldStartLoadWithRequest(event) {
        // Implement any custom loading logic here, don't forget to return!
        return true;
    }

    componentDidMount(){ }

    webGoBack(){
        // console.log(this.refs[WEBVIEW_REF]);
        // this.goBack()
        this.refs[WEBVIEW_REF].goBack();
    }

    onNavigationStateChange(navState) {
        this.setState({
            backButtonEnabled: navState.canGoBack,
            forwardButtonEnabled: navState.canGoForward,
            url: navState.url,
            status: navState.title,
            loading: navState.loading,
            scalesPageToFit: true,

            onlyWeb: this.props.onlyWeb||false,
        });
    }

    render(){
        H5Back = this;
        return (
            <View style={{flex: 1}}>
                {(
                    ()=>{
                        if (!this.state.onlyWeb){
                            return (
                                <TouchableOpacity
                                    style={[styles.innerBg, styles.navBar]}
                                    underlayColor="rgb(199, 199, 199)"
                                    onPress={() => {
                                        _page.back()
                                    }} >
                                    <View style={styles.listView}>
                                        <Icon name="ios-arrow-back" size={30} color="rgb(13, 13, 13)"/>
                                    </View>
                                </TouchableOpacity>
                            )
                        }
                    }
                )()}

                <WebView
                  ref={WEBVIEW_REF}
                  automaticallyAdjustContentInsets={false}
                  style={styles.webView}
                  source={{uri: this.state.url}}
                  javaScriptEnabled={true}
                  domStorageEnabled={true}
                  decelerationRate="normal"
                  onNavigationStateChange={this.onNavigationStateChange}
                  onShouldStartLoadWithRequest={this.onShouldStartLoadWithRequest}
                  startInLoadingState={true}
                  scalesPageToFit={this.state.scalesPageToFit}
                />
                {( ()=>{ if (this.state.onlyWeb &&
                            this.state.url.indexOf('?topic=')>-1
                        ){
                        return (
                            <TouchableOpacity
                                style={[styles.goback]}
                                onPress={this.webGoBack}
                            >
                                <Image source={images.arrowLeft} style={{width:48, height:48}}/>
                            </TouchableOpacity>
                        )
                }} )()}
            </View>
        )
    }
}

export let wrender = function(data){
    return <Aaa data={data} onlyWeb={true}/>
}

export default function(id){
    return page.new({
        trigger: function(self, intent){
            _page = this;
            return (
                <Aaa data={intent} />
            )
        }
    })
}
