// js/scrolltab/index.js

import React from 'react';
import {
  StyleSheet,
  Text,
  View,
  ScrollView,
  TouchableOpacity,
  TouchableHighlight
} from 'react-native';

import Icon from 'react-native-vector-icons/Ionicons';
import ScrollableTabView, { DefaultTabBar, } from 'react-native-scrollable-tab-view';



/*  sample 1 */

export let simpleScrollTab = React.createClass({
  render() {
    return (
      <ScrollableTabView>
          <Text tabLabel='Tab #1'>My</Text>
          <Text tabLabel='Tab #2'>favorite</Text>
          <Text tabLabel='Tab #3'>project</Text>
      </ScrollableTabView>
    );
  }
});




/*  sample 2 */
// Using tabBarPosition='overlayTop' or 'overlayBottom' lets the content show through a
// semitransparent tab bar. Note that if you build a custom tab bar component, its outer container
// must consume a 'style' prop (e.g. <View style={this.props.style}) to support this feature.
export let defaultScrollTab = React.createClass({
  render() {
    return (
        <ScrollableTabView
          style={styles.container}
          renderTabBar={()=><DefaultTabBar backgroundColor='rgba(255, 255, 255, 0.7)' />}
          tabBarPosition='overlayTop'
        >
          <ScrollView tabLabel='iOS'>
            <Icon name='logo-apple' color='black' size={300} style={styles.icon} />
            <Icon name='ios-phone-portrait' color='black' size={300} style={styles.icon} />
            <Icon name='logo-apple' color='#DBDDDE' size={300} style={styles.icon} />
            <Icon name='ios-phone-portrait' color='#DBDDDE' size={300} style={styles.icon} />
          </ScrollView>
          <ScrollView tabLabel='Android'>
            <Icon name='logo-android' color='#A4C639' size={300} style={styles.icon} />
            <Icon name='logo-android' color='black' size={300} style={styles.icon} />
            <Icon name='logo-android' color='brown' size={300} style={styles.icon} />
          </ScrollView>
        </ScrollableTabView>
    )
  },
});



/*   sample 3 */


const FacebookTabBar = React.createClass({
  tabIcons: [],

  propTypes: {
    goToPage: React.PropTypes.func,
    activeTab: React.PropTypes.number,
    tabs: React.PropTypes.array,
  },

  componentDidMount() {
    this._listener = this.props.scrollValue.addListener(this.setAnimationValue);
  },

  setAnimationValue({ value, }) {
    this.tabIcons.forEach((icon, i) => {
      const progress = (value - i >= 0 && value - i <= 1) ? value - i : 1;
      icon.setNativeProps({
        style: {
          color: this.iconColor(progress),
        },
      });
    });
  },

  //color between rgb(59,89,152) and rgb(204,204,204)
  iconColor(progress) {
    const red = 59 + (204 - 59) * progress;
    const green = 89 + (204 - 89) * progress;
    const blue = 152 + (204 - 152) * progress;
    return `rgb(${red}, ${green}, ${blue})`;
  },

  render() {
    return (
        <View style={[styles.tabs, this.props.style, ]}>
            {
                this.props.tabs.map((tab, i) => {
                    return (
                        <TouchableOpacity key={tab} onPress={() => this.props.goToPage(i)} style={styles.tab}>
                            <Icon
                                name={tab}
                                size={30}
                                color={this.props.activeTab === i ? 'rgb(59,89,152)' : 'rgb(204,204,204)'}
                                ref={(icon) => { this.tabIcons[i] = icon; }}
                            >
                            </Icon>
                            <Text style={{fontSize:14}}>1234</Text>
                        </TouchableOpacity>
                    )
                })
            }
        </View>
    )
  },
});

export let iconScrollTab =  React.createClass({
  render() {
    return (
        <ScrollableTabView
          style={{marginTop: 30, }}
          initialPage={1}
          renderTabBar={() => <FacebookTabBar />} >

          <ScrollView tabLabel="ios-paper" style={styles.tabView}>
            <View style={styles.card}>
              <Text>News</Text>
            </View>
          </ScrollView>
          <ScrollView tabLabel="ios-people" style={styles.tabView}>
            <View style={styles.card}>
              <Text>Friends</Text>
            </View>
          </ScrollView>
          <ScrollView tabLabel="ios-chatboxes" style={styles.tabView}>
            <View style={styles.card}>
              <Text>Messenger</Text>
            </View>
          </ScrollView>
          <ScrollView tabLabel="ios-notifications" style={styles.tabView}>
            <View style={styles.card}>
              <Text>Notifications</Text>
            </View>
          </ScrollView>
          <ScrollView tabLabel="ios-list" style={styles.tabView}>
            <View style={styles.card}>
              <Text>Other nav</Text>
            </View>
          </ScrollView>

        </ScrollableTabView>
    )
  },
});

const styles = StyleSheet.create({
    container: {
      marginTop: 30,
    },
    icon: {
      width: 300,
      height: 300,
      alignSelf: 'center',
    },

  tabView: {
    flex: 1,
    padding: 10,
    backgroundColor: 'rgba(0,0,0,0.01)',
  },
  card: {
    borderWidth: 1,
    backgroundColor: '#fff',
    borderColor: 'rgba(0,0,0,0.1)',
    margin: 5,
    height: 150,
    padding: 15,
    shadowColor: '#ccc',
    shadowOffset: { width: 2, height: 2, },
    shadowOpacity: 0.5,
    shadowRadius: 3,
  },

  tab: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    paddingBottom: 10,
  },
  tabs: {
    height: 45,
    flexDirection: 'row',
    paddingTop: 5,
    borderWidth: 1,
    borderTopWidth: 0,
    borderLeftWidth: 0,
    borderRightWidth: 0,
    borderBottomColor: 'rgba(0,0,0,0.05)',
  },
});
