// js/index/index.js
/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 * @flow
 */

import React, { Component } from 'react';
import styles from 'css/index';
import FtabBar from 'comp/modules/tabbar';
import FlistView, {pure as PureList, store} from 'comp/modules/listview';
import { View, Text, Image, TouchableHighlight } from 'react-native';


const getListData = () => {
    return {
        url: apis.dbdemo
    }
}

function dealWithListData(data){
    let _data = []
    if (data && data.length){
        data.map( (item, i)=>{
            _data.push({ title: (
                <TouchableHighlight
                    key={'titem'+i}
                    style={styles.innerBg}
                    underlayColor="rgb(199, 199, 199)"
                    onPress={() => {
                        console.log('=======');
                    }}
                >
                    <View style={styles.inner}>
                        <Image
                        source={{uri: item.img||'http://localhost:3000/images/logo128.png'}}
                        style={styles.thumbnail}
                        />
                        <View style={styles.rightPart}>
                            <Text style={styles.title}>{item.title.substring(0, 6)}</Text>
                            <Text style={styles.year}>{item.create_at}</Text>
                            <Text style={styles.subTitle}>{item.title.substring(0,14)}</Text>
                        </View>
                        <Image source={ images.leftTile } style={styles.dot} />
                    </View>
                </TouchableHighlight>
            )})
        })
    }
    return _data;
}

// 异步拿去数据
async function asyData(){
    let _data = await req( apis.dbdemo );
    let __data = dealWithListData(_data);
    SAX.setter('PurePure', {data: __data})    // 重写listView
}


// 列表数据
let blogList = (type) => {
    if (type===1){  // store list
        let StoreListView = store(FlistView);
        return <StoreListView name='Xxx' data={ getListData() } />
    }
    else
    if (type===2){  // with url list，自己跑异步数据
        return <FlistView data={ getListData() } />
    }
    else {  // pure list , 把异步数据抽离，并通过 SAX re render 列表组件
        let StorePureListView = store(PureList);
        return <StorePureListView name='PurePure' data={[]} />
    }

}

// tabbar数据
function tabBarData(){
    return [
        {title: 'facemash', content: blogList(3)},
        {title: 'messages'},
        {title: '设置', icon: images.inner['settings']},
    ]
}

// for tabbar 回调方法
// 每一个item
function tabBaritemMethod(connect, selectedTab, tabItems){
    let _this = this;
    connect({
        onPress: function(event){
            if (event.target === tabItems[0]){
                selectedTab('facemash')
                asyData()
            }
            if (event.target === tabItems[1]){
                selectedTab('messages')
            }
            if (event.target === tabItems[2]){
                selectedTab('设置')
            }
        }
    })
}

// 输出结构
export default class hello extends Component {
    constructor(props){
        super(props);
    }

    render() {
        let dataTabBar = tabBarData();
        return (
            <View style={{flex: 1,}}>
                <FtabBar data={dataTabBar} itemMethod={tabBaritemMethod}/>
            </View>
        )
    }
}
