// js/index/index3.js
import React, { Component } from 'react';
import { View, Text, Image, TouchableHighlight } from 'react-native';
import styles from 'css/index';
import tabs from 'comp/modules/tabbar/test';
import Icon from 'react-native-vector-icons/Ionicons';


// tags item button
let blogIcon = [
    <View>
        <Icon name="ios-home-outline" size={30} color="#4F8EF7" />
        <Text style={{fontSize:11}}>博客</Text>
    </View>,
    <View>
        <Icon name="ios-home" size={30} color="#4F8EF7" />
        <Text style={{fontSize:11}}>博客</Text>
    </View>
]

let msgIcon = [
    <View>
        <Icon name="ios-chatbubbles-outline" size={30} color="#4F8EF7" />
        <Text style={{fontSize:11}}>消息</Text>
    </View>,
    <View>
        <Icon name="ios-chatbubbles" size={30} color="#4F8EF7" />
        <Text style={{fontSize:11}}>消息</Text>
    </View>
]

let setIcon = [
    <View>
        <Icon name="ios-settings-outline" size={30} color="#4F8EF7" />
        <Text style={{fontSize:11}}>设置</Text>
    </View>,
    <View>
        <Icon name="ios-settings" size={30} color="#4F8EF7" />
        <Text style={{fontSize:11}}>设置</Text>
    </View>
]

let bdgIcon = [
    <View>
        <Icon name="ios-settings-outline" size={30} color="#4F8EF7" />
        <Text style={{fontSize:11}}>红点</Text>
    </View>,
    <View>
        <Icon name="ios-settings" size={30} color="#4F8EF7" />
        <Text style={{fontSize:11}}>红点</Text>
    </View>
]

let rtIcon = [
    <View>
        <Icon name="ios-settings-outline" size={30} color="#4F8EF7" />
        <Text style={{fontSize:11}}>路由</Text>
    </View>,
    <View>
        <Icon name="ios-settings" size={30} color="#4F8EF7" />
        <Text style={{fontSize:11}}>路由</Text>
    </View>
]


// tags item config
let _blog = {
    title: blogIcon,
    content: require('./business/blog').default,
    method:()=>{}
}

let _msg = {
    title: msgIcon,
    content: 'content为字符串会自动在tabs里生成结构',
    method:()=>{
        console.log('消息中心')
        console.log(SAX.lister());
    }
}

let _set = {
    title: setIcon,
    content: (<View>
            <Text>content为自定义结构，会自动输出</Text>
        </View>
    ),
    method:()=>{
        console.log('设置中心')
    }
}

let _bdg = {
    title: bdgIcon,
    content: '可以有badge',
    badge: 1
}


export default tabs(
    // tabs config
    {
        blog: _blog,
        message: _msg,
        setting: _set,
        badge: _bdg
    },

    // options
    {
        sax: 'TabBlog'
    }
)
