// js/index/index2.js
/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 * @flow
 */
import React, { Component } from 'react';
import styles from 'css/index';
import tabs from 'comp/modules/tabbar/test';
import {pure as PureList, store} from 'comp/modules/listview';
import { View, Text, Image, TouchableHighlight } from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';

function dealWithListData(data){
    let _data = []
    if (data && data.length){
        data.map( (item, i)=>{
            _data.push({ title: (
                <TouchableHighlight
                    key={'titem'+i}
                    style={styles.innerBg}
                    underlayColor="rgb(199, 199, 199)"
                    onPress={() => {
                        console.log('========== ');
                        SAX.setter('SaxTabs', {data:{badge: {blog: 3} }})  // 更改badge数据，用于websocket动态更新badge
                    }}>

                    <View style={styles.inner}>
                        <Image
                        source={{uri: item.img||'http://localhost:3000/images/logo128.png'}}
                        style={styles.thumbnail}
                        />
                        <View style={styles.rightPart}>
                            <Text style={styles.title}>{item.title.substring(0, 6)}</Text>
                            <Text style={styles.year}>{item.create_at}</Text>
                            <Text style={styles.subTitle}>{item.title.substring(0,14)}</Text>
                        </View>
                        <Image source={ images.leftTile } style={styles.dot} />
                    </View>

                </TouchableHighlight>
            )})
        })
    }
    return _data;
}

// 列表数据
function blogList(type){
    // 异步数据
    async function asyData(){
        let _data = await req( apis.dbdemo );
        let __data = dealWithListData(_data);
        SAX.setter('PurePure', {data: __data})    // 重写listView
    }

    setTimeout(()=>{
        asyData()
    }, 0);

    let StorePureListView = store(PureList);
    return <StorePureListView name='PurePure'/>
}

let blogIcon = [
    <View>
        <Icon name="ios-home-outline" size={30} color="#4F8EF7" />
        <Text style={{fontSize:11}}>博客</Text>
    </View>,
    <View>
        <Icon name="ios-home" size={30} color="#4F8EF7" />
        <Text style={{fontSize:11}}>博客</Text>
    </View>
]

let msgIcon = [
    <View>
        <Icon name="ios-chatbubbles-outline" size={30} color="#4F8EF7" />
        <Text style={{fontSize:11}}>消息</Text>
    </View>,
    <View>
        <Icon name="ios-chatbubbles" size={30} color="#4F8EF7" />
        <Text style={{fontSize:11}}>消息</Text>
    </View>
]

let setIcon = [
    <View>
        <Icon name="ios-settings-outline" size={30} color="#4F8EF7" />
        <Text style={{fontSize:11}}>设置</Text>
    </View>,
    <View>
        <Icon name="ios-settings" size={30} color="#4F8EF7" />
        <Text style={{fontSize:11}}>设置</Text>
    </View>
]

// tabs 带 pageView
export default tabs({
    blog: {title: blogIcon, content: blogList(), badge: 11},
    message: {title: msgIcon, content: '消息中心', method: function(){console.log(1234)} },
    setting: {title: setIcon, content: '设置中心', method: function(){console.log(4455)} },
}, {
    sax: 'SaxTabs'
})
