// comp/widgets/Anchor.js
'use strict';

var React = require('react');
var ReactNative = require('react-native');
var {
  Linking,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
} = ReactNative;

let OpenURLButton = React.createClass({

  propTypes: {
    href: React.PropTypes.string,
  },

  handleClick: function() {
    Linking.canOpenURL(this.props.href).then(supported => {
      if (supported) {
        Linking.openURL(this.props.href);
      } else {
        console.log('Don\'t know how to open URI: ' + this.props.href);
      }
    });
  },

  // <Text style={styles.text}>Open {this.props.href}</Text>
  render: function() {
    return (
      <TouchableOpacity
        onPress={this.handleClick}>
        <View style={styles.button}>
            {this.props.children}
        </View>
      </TouchableOpacity>
    );
  }
});

export default OpenURLButton;

var styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'white',
    padding: 10,
    paddingTop: 30,
  },
  button: {
    marginTop: 10,
  },
  text: {
    color: 'white',
  },
});
